### ISSUE TITLE
Setting array state with empty array results in array with empty string

### ISSUE THREAD

**Issue by franzwilhelm** [47ng/next-usequerystate#329]
I have this code
```ts
  const [locationIds, setLocationIds] = useQueryState(
    'locationIds',
    queryTypes.array(queryTypes.string).withDefault([]),
  );
```
When calling this
```ts
setLocationIds([]);
```
It results in an empty query: `?locationIds=`, and the state is populated with `['']`. I worked around it by wrapping the setState with a check, but it's suboptimal. When setting an empty array, it should not be populated with an empty string

**Comment by prasannamestha** [comment]
Facing the same issue. Does anybody know why this is happening?

**Comment by franky47** [comment]
The issue is in the array parser:
https://github.com/47ng/next-usequerystate/blob/aa9609841c8d9b1d09efe76513586f5db8c5c448/src/defs.ts#L197

`"".split(',')` returns `['']`, so to properly handle the empty array case, a test should be performed on the query value (once trimmed) to see if it's an empty string, and if so return an empty array explicitly.

**Comment by prasannamestha** [comment]
> The issue is in the array parser:
> 
> https://github.com/47ng/next-usequerystate/blob/aa9609841c8d9b1d09efe76513586f5db8c5c448/src/defs.ts#L197
> 
> `''.split(',')` returns `['']`, so to properly handle the empty array case, a test should be performed on the query value (once trimmed) to see if it's an empty string, and if so return an empty array explicitly.
> 
> Would you like to open a PR?

Sure, sending in a PR

**Comment by franzwilhelm** [comment]
Thanks guys!

**Comment by franky47** [comment]
Fixed in #330 by @prasannamestha, thanks!

**Comment by franky47** [comment]
Fix is shipped in version 1.7.3.

**Comment by franzwilhelm** [comment]
🙌
